Lib/ldap/ldapobject.py

```diff
@@ -298,14 +297,15 @@ def _ldap_call(self,func,*args,**kwargs):
       finally:
         self._ldap_object_lock.release()
     except LDAPError as e:
+      exc_type,exc_value,exc_traceback = sys.exc_info()
```

Comment:
exc_info() is problematic. It may introduce reference cycles that keep frames alive for a long time. The function is especially problematic in Python 3 because of __traceback__ info. 

* https://pagure.io/freeipa/issue/6696
* http://engineering.hearsaysocial.com/2013/06/16/circular-references-in-python/
* https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/